runtime.go

```diff
@@ -63,6 +64,14 @@ type _runtime struct {
 
 func (self *_runtime) enterScope(scope *_scope) {
 	scope.outer = self.scope
+	if self.scope != nil {
+		scope.depth = self.scope.depth + 1
+	}
+
+	if self.stackLimit != 0 && scope.depth >= self.stackLimit {
```

Comment:
As an optimisation would it make sense to not do this check if the above if failed?